lib/blackout_updater.rb

```diff
@@ -0,0 +1,37 @@
+class BlackoutUpdater
+  # Service Object to update blackouts in the blackouts controller
+  def initialize(blackout:, update_p:)
+    @blackout = blackout
+    @update_p = update_p
+  end
+
+  def update!
+    return { result: nil, error: error } if error
+    update_handler
+  end
+
+  private
+
+  attr_reader :blackout, :update_p
+
+  def error
+    conflicts = Reservation.overlaps_with_date_range(update_p[:start_date],
+                                                     update_p[:end_date]).active
+    return unless conflicts.any?
+    list_conflicts(conflicts)
+  end
+
+  def list_conflicts(conflicts)
+    message = 'The following reservation(s) will be unable to be returned: '
+    conflicts.each do |conflict|
+      message += "#{conflict.md_link}, "
+    end
+    return message[0, message.length - 2]\
+           + '. Please update their due dates and try again.'
+  end
```

Comment:
I think the processing of the conflicts should be pulled out to a separate method so that this method reads: `"The following reservation(s) will be unable to be returned #{conflicts_string}. Please update their due dates and try again."`